steam_deploy.sh

```diff
@@ -54,15 +54,27 @@ echo "#    Generating App Manifest    #"
 echo "#################################"
 echo ""
 
-mkdir BuildOutput
+mkdir -p BuildOutput
+
+steamdir=$STEAM_HOME
+manifest_path=$(pwd)/manifest.vdf
+contentroot=$(pwd)/$rootPath
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  steamdir="$HOME/Library/Application Support/Steam"
+elif [[ "$OSTYPE" == "msys"* ]]; then
+  manifest_path=$(cygpath -w "$manifest_path")
+  contentroot=$(cygpath -w "$contentroot")
+elif [ "$RUNNER_OS" == "Linux" ]; then
```

Comment:
Tested and got an `unexpected operator`: https://github.com/finol-digital/Card-Game-Simulator/actions/runs/3120532644

Looks like string comparison should be done with `=` instead of `==`.

```suggestion
elif [ "$RUNNER_OS" = "Linux" ]; then
```